rpcs/httpFetcher.go

```diff
@@ -45,15 +45,16 @@ type FetcherClient interface {
 type HTTPFetcher struct {
 	peer    network.HTTPPeer
 	rootURL string
+	net     *network.GossipNode
```

Comment:
I'm pretty confident that you don't need the `*network.GossipNode` but rather the `network.GossipNode`. The latter is already an interface, so it's a struct pointer.